src/plugins/discover/public/application/main/discover_main_route.tsx

```diff
@@ -109,7 +105,7 @@ export function DiscoverMainRoute({
       const hasUserDataViewValue = await data.dataViews.hasData
         .hasUserDataView()
         .catch(() => false);
-      const hasESDataValue = isDev || (await data.dataViews.hasData.hasESData().catch(() => false));
+      const hasESDataValue = await data.dataViews.hasData.hasESData().catch(() => false);
```

Comment:
`@elastic/kibana-data-discovery`, hope you don't mind. Because of `isDev` check I couldn't add a functional test for this state that would pass locally. This also made it harder to discover the bug

Reply:
I think this is fine, its was introduced to resolve https://github.com/elastic/kibana/issues/133273
Implemented in https://github.com/elastic/kibana/pull/133461

But I now tested what was mentioned not to work in #133273, so I think it has been resolved differently in the meantime? 

It seems that `isDev` is now deprecated, which means we should cleanup in a follow up